crates/nu-command/src/formats/to/nuon.rs

```diff
@@ -189,26 +196,22 @@ fn to_nuon(call: &Call, input: PipelineData) -> Result<String, ShellError> {
     value_to_string(&v, call.head)
 }
 
+lazy_static! {
+    // This case-insensitive regex hits:
+    // • Any character of []:`{}#';()|$,
+    //   • The above hits Datetimes via hitting :
+    // • Any whitespace
+    // • true, false, null
+    // • inf, Infinity, nan
+    // • Integers and floats
+    // • Filesizes
+    // • Durations
+    // • Ranges
+    static ref NEED_QUOTE_REGEX: Regex = Regex::new(r#"(?i)[\[\]:`\{\}#'";\(\)\|\$,\s]|^(true|false|null|[+-]?inf(inity)?|[+-]?nan)$|^[+-]?\d+\.?\d*(([kmgtpez]?i?b)|[mnu]s|sec|min|hr|day|wk)?$|^([+-]?\d+\.?\d*)?\.\.<?([+-]?\d+\.?\d*)?$"#).expect("internal error: NEED_QUOTE_REGEX didn't compile");
```

Comment:
Wow that's a mouthful of a regex :D
Do we need a `fancy_regex` to do this? If we need backreferences, in the worst case you might be able to craft DoS strings.
The other question would be if we would be willing to accept a simpler heuristic with some false positives if this turns out to be a bottleneck in emitting nuon.

Reply:
This is already relatively simple (it false-positives on various invalid ranges like `..<` and invalid capitalisations like `TRUE`, so that it can hit every capitalisation of `nan`, `infinity`, filesize suffixes etc.).

This has no backreferences so I could presumably turn it into a "vanilla" regex.

Reply:
If I remember correctly `fancy_regex` builds on top of `regex` so pulling this in might not be an additional dep.

But optimization of the performance can be a second step, if we start to benchmark this codepath.

Reply:
I think @sholderbach is right and I think that we should prefer `regex` over `fancy_regex` everywhere possible, and I think it would be nice -- if this regex doesn't use `fancy_regex` features -- to express it using `regex` because that would prove that it doesn't have any of the undesirable runtime complexities that `fancy_regex` allows.

Reply:
Well… `fancy-regex` does "build upon" `regex`, but as a fork of `regex` rather than something that uses it as an explicit dep. It does [promise](https://docs.rs/fancy-regex/latest/fancy_regex/) to have the same performance of `regex` when none of [this syntax](https://docs.rs/fancy-regex/latest/fancy_regex/#syntax) is in the regex… and, like, you can just read the regex and see none of that stuff appears in it. To me, explicitly bringing in the `regex` crate doesn't seem like it would accomplish much.

Reply:
To be clear, when I was researching `fancy-regex` if not "fancy" features are used, then it uses regex by default.